src/SI_Toolkit/Predictors/predictor_wrapper.py

```diff
@@ -0,0 +1,136 @@
+        self.predictor_config = dcp(self.predictors_config[self.predictor_name])
+        self.predictor_type: str = self.predictor_config['predictor_type']
+        self.model_name: str = self.predictor_config['model_name']
+
+
+    def configure(self, batch_size, horizon, predictor_specification=None, compile_standalone=False):
+
+        self.update_predictor_config_from_specification(predictor_specification)
+
+        compile_standalone = {'disable_individual_compilation': not compile_standalone}
+
+        self.batch_size = batch_size
+        self.horizon = horizon
+
+        if self.predictor_type == 'neural':
+            from SI_Toolkit.Predictors.predictor_autoregressive_tf import predictor_autoregressive_tf
+            self.predictor = predictor_autoregressive_tf(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        elif self.predictor_type == 'GP':
+            from SI_Toolkit.Predictors.predictor_autoregressive_GP import predictor_autoregressive_GP
+            self.predictor = predictor_autoregressive_GP(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        elif self.predictor_type == 'ODE':
+            from SI_Toolkit.Predictors.predictor_ODE import predictor_ODE
+            self.predictor = predictor_ODE(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config)
+
+        elif self.predictor_type == 'ODE_TF':
+            from SI_Toolkit.Predictors.predictor_ODE_tf import predictor_ODE_tf
+            self.predictor = predictor_ODE_tf(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        else:
+            raise NotImplementedError('Type of the predictor not recognised.')
+
+    def configure_with_compilation(self, batch_size, horizon, predictor_specification=None):
+        """
+        To get max performance
+        use this for standalone predictors (like in Brunton test)
+        but not predictors within controllers
+        """
+        self.configure(batch_size, horizon, compile_standalone=True)
+
+    def update_predictor_config_from_specification(self, predictor_specification: str = None):
+
+        if predictor_specification is None:  # The default values are not modified
+            return
+        if predictor_specification == 'default':
+            self.predictor_name: str = self.predictor_name_main
+            self.predictor_config = dcp(self.predictors_config['predictors'][self.predictor_name])
+            self.predictor_type: str = self.predictor_config['predictor_type']
+            self.model_name: str = self.predictor_config['model_name']
+
+
+        predictor_name = None
+        model_name = None
+        model_name_contains_path_to_model = False
+
+        # Sort out predictor name from component specification
+        predictor_specification_components = predictor_specification.split(":")
+
```

Comment:
Perhaps assert here that the `predictor_specification_components` have a certain length, in case there is a fixed expectation. But only optional

Reply:
No, they are flexible to have various length. Eg. "GRU-1IN-2H1-1OUT", "I_love_control", "ODE" or "ODE_TF:simple" - in the last case it takes ODE_TF_default from config_predictors and overwrites model_name with "simple"

Reply:
Perfect